Main/OOP18/drink_factory.cpp

```diff
@@ -0,0 +1,16 @@
+#include "drink_factory.h"
+
+Drink* createDrink(DrinkType type, SizeType size) {
+    switch (type) {
+    case DrinkType::ESPRESSO:
+    case DrinkType::AMERICANO:
+    case DrinkType::LATTE:
+    case DrinkType::CAPPUCCINO:
+        return new Coffee(type, size);
+    case DrinkType::BLACK_TEA:
+    case DrinkType::GREEN_TEA:
+        return new Tea(type, size);
```

Comment:
same here: create TeaType enum, convert DrinkType into TeaType and pass it instead into c-tor

Reply:
Feel free to ping me in slack if it's not clear enough why you need 3 enum classes for types